scripts/pre_commit_linter.py

```diff
@@ -1463,6 +1463,97 @@ def _check_directive_scope(all_files):
     return summary_messages
 
 
+def _check_js_property(all_files):
+    """This function checks that all directives have an explicit
+    scope: {} and it should not be scope: true.
+    """
+    print 'Starting js property check'
+    print '----------------------------------------'
+    # Select JS files which need to be checked.
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js') and not filename.endswith('app.js') and not
+        filename.endswith('directives.js')]
+    failed = False
+    summary_messages = []
+    property_name = ''
+    for filename in files_to_check:
+        property_num = 0
+        # Filename without its path.
+        exact_filename = filename.split('/')[-1][:-3]
+        content = FileCache.read(filename)
+        parsed_dict = _validate_and_parse_js_file(filename, content)
+        # Parse the body of the content as nodes.
+        parsed_nodes = parsed_dict['body']
+        for parsed_node in parsed_nodes:
+            # Check the type of the node.
+            if parsed_node['type'] != 'ExpressionStatement':
+                continue
+            # Separate the expression part of the node.
+            expression = parsed_node['expression']
+            # Check whether the expression belongs to a directive.
+            expression_type_is_not_call = (
+                expression['type'] != 'CallExpression')
+            if expression_type_is_not_call:
+                continue
+            expression_callee_type_is_not_member = (
+                expression['callee']['type'] != 'MemberExpression')
+            if expression_callee_type_is_not_member:
+                continue
+            callee_property = expression['callee']['property']['name']
+            if callee_property != 'factory' and (
+                    callee_property != 'directive' and
+                    callee_property != 'controller'):
```

Comment:
There should be a check for filters as well (In rule of 1 [here](https://angular.io/guide/upgrade#follow-the-angularjs-style-guide), it is specified to have each controller, component, service, and filter is in its own source file.)

Reply:
Done